shift/src/main/java/org/coursera/android/shift/ShiftPref.java

```diff
@@ -13,14 +13,38 @@
 
 package org.coursera.android.shift;
 
-interface ShiftPref<T> {
-    T getValue();
+abstract class ShiftPref <T> {
+    protected final ShiftPersistenceManager PERSISTENCE;
+    protected final String KEY;
+    protected final T DEFAULT_VALUE;
+    protected final Class<T> CLASS;
 
-    void setValue(T value);
+    public ShiftPref(ShiftPersistenceManager persistenceManager,
```

Comment:
Turned ShiftPref into an abstract class so I could reuse code for isValueSet, delete, and setValueToDefault